engine/access/ingestion2/execution_receipts_consumer.go

```diff
@@ -0,0 +1,133 @@
+package ingestion2
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/common/fifoqueue"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/storage"
+)
+
+const defaultQueueCapacity = 10_000
+
+type ExecutionReceiptConsumer struct {
+	log                      zerolog.Logger
+	collectionExecutedMetric module.CollectionExecutedMetric
+
+	messageHandler       *engine.MessageHandler
+	receiptsMessageQueue engine.MessageStore
+	notifier             engine.Notifier
+	receipts             storage.ExecutionReceipts
+
+	errorMessageRequester ErrorMessageRequester
+}
+
+func NewExecutionReceiptConsumer(
+	log zerolog.Logger,
+	collectionExecutedMetric module.CollectionExecutedMetric,
+	receipts storage.ExecutionReceipts,
+	errorMessageRequester ErrorMessageRequester,
+) (*ExecutionReceiptConsumer, error) {
+	receiptsRawQueue, err := fifoqueue.NewFifoQueue(defaultQueueCapacity)
+	if err != nil {
+		return nil, fmt.Errorf("could not create execution receipts queue: %w", err)
+	}
+
+	receiptsMessageQueue := &engine.FifoMessageStore{FifoQueue: receiptsRawQueue}
+	messageHandler := engine.NewMessageHandler(
+		log,
+		engine.NewNotifier(),
+		engine.Pattern{
+			Match: func(msg *engine.Message) bool {
+				_, ok := msg.Payload.(*flow.ExecutionReceipt)
+				return ok
+			},
+			Store: receiptsMessageQueue,
+		},
+	)
+
+	return &ExecutionReceiptConsumer{
+		log:                      log,
+		collectionExecutedMetric: collectionExecutedMetric,
+		receipts:                 receipts,
+		messageHandler:           messageHandler,
+		notifier:                 engine.NewNotifier(),
+		errorMessageRequester:    errorMessageRequester,
+	}, nil
+}
+
+// Notify processes a new execution receipt payload and notifies the consumer to begin processing queued messages.
+//
+// No errors are expected during normal operations.
+func (c *ExecutionReceiptConsumer) Notify(originID flow.Identifier, payload interface{}) error {
+	err := c.messageHandler.Process(originID, payload)
+	c.notifier.Notify()
+	return err
```

Comment:
which means you can get rid or the call to notify here.

Consider defining an interface with 
```
Process(flow.Identifier, interface{}) error
```

then embedding it in this struct. then you could pass the message handler as the implementation and skip this method entirely